com.unity.ml-agents/Runtime/Analytics/InferenceAnalytics.cs

```diff
@@ -0,0 +1,307 @@
+            // Note - to debug, use JsonUtility.ToJson on the event.
+            //Debug.Log(JsonUtility.ToJson(data, true));
+#if UNITY_EDITOR
+            // TODO re-enable when we're ready to merge.
+            //EditorAnalytics.SendEventWithLimit(k_EventName, data);
+#else
+            return;
+#endif
+        }
+
+
+        /// <summary>
+        /// Generate an InferenceEvent for the model.
+        /// </summary>
+        /// <param name="nnModel"></param>
+        /// <param name="behaviorName"></param>
+        /// <param name="inferenceDevice"></param>
+        /// <param name="sensors"></param>
+        /// <param name="actionSpec"></param>
+        /// <returns></returns>
+        static InferenceEvent GetEventForModel(
+            NNModel nnModel,
+            string behaviorName,
+            InferenceDevice inferenceDevice,
+            IList<ISensor> sensors,
+            ActionSpec actionSpec
+        )
+        {
+            var barracudaModel = ModelLoader.Load(nnModel);
+            var inferenceEvent = new InferenceEvent();
+            inferenceEvent.BehaviorName = behaviorName;
+            inferenceEvent.BarracudaModelSource = barracudaModel.IrSource;
+            inferenceEvent.BarracudaModelVersion = barracudaModel.IrVersion;
+            inferenceEvent.BarracudaModelProducer = barracudaModel.ProducerName;
+            inferenceEvent.MemorySize = (int)barracudaModel.GetTensorByName(TensorNames.MemorySize)[0];
+            inferenceEvent.InferenceDevice = (int)inferenceDevice;
+
+            if (barracudaModel.ProducerName == "Script")
+            {
+                // .nn files don't have these fields set correctly. Assign some placeholder values.
+                inferenceEvent.BarracudaModelSource = "NN";
+                inferenceEvent.BarracudaModelProducer = "tf2bc.py";
+            }
+
+#if UNITY_2019_3_OR_NEWER
+            var barracudaPackageInfo = UnityEditor.PackageManager.PackageInfo.FindForAssembly(typeof(Tensor).Assembly);
+            inferenceEvent.BarracudaPackageVersion = barracudaPackageInfo.version;
+#else
+            inferenceEvent.BarracudaPackageVersion = "unknown";
+#endif
+
+            inferenceEvent.ActionSpec = EventActionSpec.FromActionSpec(actionSpec);
+            inferenceEvent.ObservationSpecs = new List<EventObservationSpec>(sensors.Count);
+            foreach (var sensor in sensors)
+            {
+                inferenceEvent.ObservationSpecs.Add(EventObservationSpec.FromSensor(sensor));
+            }
+
+            inferenceEvent.TotalWeightSizeBytes = GetModelWeightSize(barracudaModel);
+            inferenceEvent.ModelHash = GetModelHash(barracudaModel);
+            return inferenceEvent;
+        }
+
+        /// <summary>
+        /// Simple implementation of the Fowler–Noll–Vo hash function.
+        /// https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function
+        /// </summary>
+        internal class FNVHash
```

Comment:
I'm not quite sure what to do about the hashing here.
* Ideally, we'd just use Hash128.Append(), which has an overload for `T[]` arguments. However, this is only available in 2020.1 and later.
* We can approximate the same behavior with HashUtilities.ComputeHash128 and HashUtilities.AppendHash, but  ComputeHash128 only works on a single element at time.
* FNVHash is a simple hash that "probably" is "good enough" for this purpose, since we're hashing a lot of data and not putting it into a hash table.

On a test with GridWorld (which has some large layers), FNV is about 5x faster 
```
FNV hash took 0.001812 seconds
hash128 took 0.007188 seconds
```
But that's also not too bad in absolute terms since it only happens once at startup.

Also note that we don't hash all the weights - if I don't limit the number per layer, the times are unusably slow:
```
hash128 hash took 6.297299 seconds
FNV hash took 1.741669 seconds
```
I'm leaning towards keeping the hash128 approach since it should be consistent between editor versions (need a test to confirm this though) and not noticeably slower (with the limit in place).

@surfnerd @vincentpierre Thoughts?

Reply:
Would it save a lot of effort to wrap this into `#if UNITY_2020_1_OR_LATER` so we only can use the latest APIs ?

Reply:
Do you mean wrap the hashing, or wrap the whole event sending? I disagree with both:
* I think the hashing is important to avoid overcounting specific models (mostly ours, but could happen from users too, e.g. the hummingbird tutorial)
* I think limiting the events to 2020+ limits the amount of data that we can get. Just knowing how many users are on what version affects decisions on whether to drop support for editor versions.

Reply:
If you're worried about the complexity, we should just remove the FMV hashing and go with the Hash128 version even if it's a bit slower.

Reply:
> I'm leaning towards keeping the hash128 approach since it should be consistent between editor versions

seems reasonable to me.

Reply:
Removed FMV, and rearranged the `#if UNITY_2020_1_OR_LATER` so that it's deeper in the implementation